blackjax/tempered_smc.py

```diff
@@ -0,0 +1,116 @@
+"""Public API for the Tempered SMC Kernel"""
+from typing import Callable, Dict, List, NamedTuple, Tuple, Union
+
+import jax
+import jax.numpy as jnp
+import numpy as np
+
+import blackjax.inference.base as base
+from blackjax.hmc import HMCParameters
+
+Array = Union[np.ndarray, jnp.DeviceArray]
+PyTree = Union[Dict, List, Tuple]
+
+__all__ = ["new_state", "kernel"]
+
+
+class SMCState(NamedTuple):
+    """ Current state for the tempered SMC algorithm. This encodes the current set of sampled particles
+    as well as the corresponding tempering parameter :math:`\lambda`
+    """
+    particles: PyTree[Array]
+    lamda: float
+
+
+class SMCInfo(NamedTuple):
+    """Additional information on the tempered SMC step.
+
+    weights: jnp.ndarray
+        The weights after the MCMC pass.
+    """
+    weights: jnp.ndarray
+
+
+new_state = base.new_hmc_state
+
+
+def _normalize(log_weights):
+    """ Normalize log-weights into weights
+    """
+    w = jnp.exp(log_weights - jnp.max(log_weights))
+    return w / w.sum()
+
+
+def kernel(potential_fn: Callable, mcmc_kernel: Callable, new_mcmc_state: Callable, resampling_method: Callable,
```

Comment:
Here I am torn between your solution of passing `mcmc_parameters` as an argument to this function, or the following:

```python
mcmc_kernel_factory = lambda potential: mcmc.new_kernel(potential_fn, parameters)

def kernel(potential_fn, mcmc_kernel_factory, new_mcmc_state, resampling_method, mcmc_step):
    raise NotImplementedError
```

Reply:
I prefer your version. It makes the interface cleaner

Reply:
And this pattern of partially applying comes back often in the library, eg for adaptation.